lapack/gonum/dorm2r.go

```diff
@@ -23,37 +23,50 @@ import "gonum.org/v1/gonum/blas"
 //
 // Dorm2r is an internal routine. It is exported for testing purposes.
 func (impl Implementation) Dorm2r(side blas.Side, trans blas.Transpose, m, n, k int, a []float64, lda int, tau, c []float64, ldc int, work []float64) {
-	if side != blas.Left && side != blas.Right {
+	left := side == blas.Left
```

Comment:
If the variable is used at more than just a couple of places, use the variable, otherwise replace it with `side == blas.Left`. Here I would leave the variable.